src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -135,7 +138,7 @@ public IDisposable On(string methodName, Type[] parameterTypes, Func<object[], o
                 try
                 {
                     Log.HubProtocol(_logger, _protocol.Name, _protocol.Version);
-                    await HandshakeAsync(startingConnectionState);
+                    await HandshakeAsync(startingConnectionState, cancellationToken);
```

Comment:
The fact that we can't pass the token to IConnection.StartAsync is an issue but not one that is relevant to this change.